modulemd/tests/test-modulemd-moduleindex.c

```diff
@@ -223,24 +224,30 @@ module_index_test_read (void)
   g_clear_pointer (&yaml_path, g_free);
   g_clear_pointer (&failures, g_ptr_array_unref);
 
-  /* The modulemd-packager definition
-   * This should fail to be read into a ModuleIndex because it provides no
-   * NSVCA information. It should only be importable via
-   * ModuleStream.read_file()
+  /* The modulemd-packager v2 definition
+   * Reading a packager document into an index can only be done by calling
+   * modulemd_read_packager_file(). Attempting to read it directly
+   * using modulemd_module_index_update_from_file() should trigger a warning
+   * and ignore the document.
    */
   yaml_path = g_strdup_printf ("%s/yaml_specs/modulemd_packager_v2.yaml",
                                g_getenv ("MESON_SOURCE_ROOT"));
+  modulemd_test_signal = 0;
+  signal (SIGTRAP, modulemd_test_signal_handler);
   ret = modulemd_module_index_update_from_file (
     index, yaml_path, TRUE, &failures, &error);
+  g_assert_cmpint (modulemd_test_signal, ==, SIGTRAP);
```

Comment:
Why are you expecting a signal here? If we are throwing a signal on unexpected input, that's going to crash DNF...